app/controllers/manageBusinesses/cease/IncomeSourceEndDateController.scala

```diff
@@ -136,28 +136,53 @@ class IncomeSourceEndDateController @Inject()(val authorisedFunctions: FrontendA
+//        val result = Redirect(redirectAction)
+//        val dataToSet = Map(
+//          CeaseIncomeSourceData.dateCeasedField -> validatedInput.date.toString,
+//          CeaseIncomeSourceData.incomeSourceIdField -> incomeSourceId.value
+//        )
+//
+//        sessionService.setMultipleMongoData(dataToSet, JourneyType(Cease, incomeSourceType)).flatMap {
+//          case Right(_) => Future.successful(result)
+//          case Left(_) => Future.failed(new Error(
+//            s"Failed to set data in session storage. incomeSourceType: $incomeSourceType."))
+//        }
+//
+//      case _ =>
+//        val propertyEndDate = validatedInput.date.toString
+//        val result = Redirect(redirectAction)
+//        sessionService.setMongoKey(key = CeaseIncomeSourceData.dateCeasedField, value = propertyEndDate,
+//          journeyType = JourneyType(Cease, incomeSourceType)).flatMap {
+//          case Right(_) => Future.successful(result)
+//          case Left(exception) => Future.failed(exception)
+//        }
+//    }
+//  }
+
+  private def handleMultipleValidatedInput(validatedInput: DateFormElement,
+                                                        incomeSourceType: IncomeSourceType,
+                                                        incomeSourceIdMaybe: Option[IncomeSourceId],
+                                                        redirectAction: Call)(
+                                                         implicit headerCarrier: HeaderCarrier): Future[Result] = {
     (incomeSourceType, incomeSourceIdMaybe) match {
       case (SelfEmployment, Some(incomeSourceId)) =>
         val result = Redirect(redirectAction)
-        sessionService.setMongoKey(
-          CeaseIncomeSourceData.dateCeasedField, validatedInput.date.toString, JourneyType(Cease, incomeSourceType)
-        ).flatMap {
-          case Right(_) =>
-            sessionService.setMongoKey(
-              CeaseIncomeSourceData.incomeSourceIdField, incomeSourceId.value, JourneyType(Cease, incomeSourceType)
-            ).flatMap {
-              case Right(_) => Future.successful(result)
-              case Left(_) => Future.failed(new Error(
-                s"Failed to set income source id in session storage. incomeSourceType: $incomeSourceType. incomeSourceType: $incomeSourceType"))
-            }
+        val dataToSet = Map(
+          CeaseIncomeSourceData.dateCeasedField -> validatedInput.date.toString,
+          CeaseIncomeSourceData.incomeSourceIdField -> incomeSourceId.value
+        )
 
+        sessionService.setMultipleMongoData(dataToSet, JourneyType(Cease, incomeSourceType)).flatMap {
+          case Right(_) => Future.successful(result)
           case Left(_) => Future.failed(new Error(
-            s"Failed to set end date value in session storage. incomeSourceType: $incomeSourceType, incomeSourceType: $incomeSourceType"))
+            s"Failed to set data in session storage. incomeSourceType: $incomeSourceType."))
```

Comment:
It'd be good to have more more detail about what has gone wrong in this case, like adding an error log which would include the date ceased

Reply:
Income source Id would be good too but i'm not sure if this is personal identifying information or not. Better to be safe than sorry and leave it out :D